src/master-playlist-controller.js

```diff
@@ -233,6 +234,7 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     this.seekable_ = videojs.createTimeRanges();
     this.hasPlayed_ = () => false;
+    this.playlistEnd = playlistEnd;
```

Comment:
It isn't necessary to attach this function to `MasterPlaylistController`, you can just call `playlistEnd` directly where you need it. On top of that, if you attach `playlistEnd` to the default export of `playlist.js`, the default export is already included in the `Hls` object available in this file. If you go that route, you do not need to import this function in a seperate import call, and instead could just call `Hls.Playlist.playlistEnd()`